frontend/pulse3d/src/components/interactiveAnalysis/WaveformGraph.js

```diff
@@ -552,31 +571,148 @@ export default function WaveformGraph({
+          let newArr = [...valleyY1];
+          newArr[wellNameToIndex[currentWell]] = y1;
+          setValleyY1(newArr);
+          newArr = [...valleyY2];
+          newArr[wellNameToIndex[currentWell]] = y2;
+          setValleyY2(newArr);
+          newArr = [...valleyYIntercept];
+          newArr[wellNameToIndex[currentWell]] = yintersept;
+          setValleyYIntercept(newArr);
+          newArr = [...valleySlope];
+          newArr[wellNameToIndex[currentWell]] = slope;
+          setValleySlope(newArr);
         }
         // descrease stroke width when unselected and dropped
         d3.select(this).attr("stroke-width", 2);
       });
+    const moveLineUpDown = d3
+      .drag()
+      .on("start", function (d) {
+        const id = d3.select(this).attr("id");
+        // close context menu if it's open
+        contextMenu.style("display", "none");
+        // increase stroke width when selected and dragging
+        d3.select(this).attr("stroke-width", 5);
+        //set starting y position
+        const yPosition = id.includes("valley")
+          ? Math.max(d.y, y(yMax + yRange))
+          : Math.min(d.y, y(yMin - yRange));
+        d3.select(this).attr("startingY", yPosition);
+      })
+      .on("drag", function (d) {
+        const id = d3.select(this).attr("id");
+
+        const yPosition = id.includes("valley")
+          ? Math.max(d.y, y(yMax + yRange))
+          : Math.min(d.y, y(yMin - yRange));
+        const initialY = d3.select(this).attr("startingY");
+        const changeInY = yPosition - initialY;
+        const newY1 = parseFloat(d3.select(this).attr("y1")) + changeInY;
+        const newY2 = parseFloat(d3.select(this).attr("y2")) + changeInY;
+
+        if (id.includes("peak")) {
+          d3.select("#peakLineY1Marker").attr("cy", newY1);
+          d3.select("#peakLineY2Marker").attr("cy", newY2);
+          d3.select(this).attr("y1", newY1);
+          d3.select(this).attr("y2", newY2);
+          d3.select(this).attr("startingY", newY1);
+        } else {
+          d3.select("#valleyLineY1Marker").attr("cy", newY1);
+          d3.select("#valleyLineY2Marker").attr("cy", newY2);
+          d3.select(this).attr("y1", newY1);
+          d3.select(this).attr("y2", newY2);
+          d3.select(this).attr("startingY", newY1);
+        }
+      })
+      .on("end", function () {
+        const id = d3.select(this).attr("id");
+        d3.select(this).attr("stroke-width", 2);
+        const y1 = y.invert(d3.select(this).attr("y1"));
+        const y2 = y.invert(d3.select(this).attr("y2"));
+        const x2 = x.invert(
+          id.includes("peak")
+            ? d3.select("#peakLineY2Marker").attr("cx")
+            : d3.select("#valleyLineY2Marker").attr("cx")
+        );
+        const slope = id.includes("peak")
+          ? peakSlope[wellNameToIndex[currentWell]]
+          : valleySlope[wellNameToIndex[currentWell]];
+        const yintersept = y2 - slope * x2;
+
+        if (id.includes("peak")) {
```

Comment:
This appears to be very similar to an if/else block I made a comment on further above. I think in addition to refactor the if/else branches to reduce duplicate code, we can also make this a function